demos/http/http_demo_s3_download_multithreaded/http_demo_s3_download_multithreaded.c

```diff
@@ -0,0 +1,1034 @@
+
+    /* Get the length of the S3 file. */
+    returnStatus = getS3ObjectFileSizeMulti( &requestInfo,
+                                             requestQueue,
+                                             responseQueue,
+                                             &fileSize );
+
+    if( returnStatus == true )
+    {
+        if( fileSize < RANGE_REQUEST_LENGTH )
+        {
+            numReqBytes = fileSize;
+        }
+        else
+        {
+            numReqBytes = RANGE_REQUEST_LENGTH;
+        }
+
+        /* Here we iterate sending byte range requests and retrieving responses
+         * until the full file has been downloaded. We keep track of the next byte
+         * to download with curByte. When this reaches the fileSize we stop
+         * downloading. We keep track of the number of responses we are waiting for
+         * with requestCount.
+         */
+        while( ( returnStatus != false ) && ( curByte < fileSize || requestCount > 0 ) )
+        {
+            /* Send range request if remaining. */
+            if( curByte < fileSize )
+            {
+                queueOpStatus = requestS3ObjectRange( &requestInfo,
+                                                      requestQueue,
+                                                      curByte,
+                                                      curByte + numReqBytes - 1 );
+
+                if( queueOpStatus == QUEUE_OP_FAILURE )
+                {
+                    returnStatus = false;
+                }
+                else if( queueOpStatus == QUEUE_OP_SUCCESS )
+                {
+                    requestCount += 1;
+                    curByte += numReqBytes;
+
+                    if( ( fileSize - curByte ) < numReqBytes )
+                    {
+                        numReqBytes = fileSize - curByte;
+                    }
+                }
+            }
+
+            /* Retrieve response. */
+            if( ( requestCount > 0 ) && ( queueOpStatus != QUEUE_OP_FAILURE ) )
+            {
+                queueOpStatus = retrieveHTTPResponse( responseQueue, &responseItem );
+
+                if( queueOpStatus == QUEUE_OP_FAILURE )
+                {
+                    returnStatus = false;
+                }
+                else if( queueOpStatus == QUEUE_OP_SUCCESS )
+                {
+                    LogInfo( ( "Main thread received HTTP response" ) );
+                    LogInfo( ( "Response Headers:\n%.*s",
+                               ( int32_t ) responseItem.response.headersLen,
+                               responseItem.response.pHeaders ) );
+                    LogInfo( ( "Response Status:\n%u", responseItem.response.statusCode ) );
+                    LogInfo( ( "Response Body:\n%.*s\n", ( int32_t ) responseItem.response.bodyLen,
+                               responseItem.response.pBody ) );
+
+                    if( responseItem.response.statusCode != 206 )
```

Comment:
One more comment that showed up in my dreams last night. Can you define this magic number as a macro? 
```
#define HTTP_STATUS_CODE_PARTIAL_CONTENT    206
```